src/pages/en/guides/cms/storyblok.md

```diff
@@ -1,14 +1,486 @@
+      <a href={post.slug}>{post.title}</a>
+      <p>{post.description}</p>
+    </li>
+  ))}
+</ul>
+```
+
+`BlogPostList.astro` is a nestable component that will render a list of blog posts previews. 
+
+It uses the `useStoryblokApi` hook to fetch all the stories with the content type of `blogPost` and the `version` query parameter to fetch the draft version of the stories when in development mode and the published version when building for production.
+
+It also adds the `storyblokEditable` directive to the parent element which will allow us to edit the blog post list in Storyblok.
+
+Finally, add your components to the `components` property of the `astro.config.mjs` file. The key is the name of the blok in Storyblok and the value is the path to the component.
+
+```js title="astro.config.mjs" ins={15-17}
+import { defineConfig } from 'astro/config';
+import storyblok from '@storyblok/astro';
+import { loadEnv } from 'vite';
+
+const env = loadEnv(import.meta.env.MODE, process.cwd(), 'STORYBLOK');
+
+export default defineConfig({
+  integrations: [
+    storyblok({
+      accessToken:
+        import.meta.env.MODE === 'development'
+          ? env.STORYBLOK_PREVIEW_TOKEN
+          : env.STORYBLOK_PUBLIC_TOKEN,
+      components: {
+        blogPost: 'storyblok/BlogPost',
+        blogPostList: 'storyblok/BlogPostList',
+        page: 'storyblok/Page',
+      },
+      apiOptions: { 
+        region: 'us',
+      },
+    })
+  ],
+});
+```
+
+### Generating pages
+
+#### Static site generation
+
+If you are using Astro's default static site generation, you will use the [dynamic routes](/en/core-concepts/routing/#dynamic-routes) and the `getStaticPaths` function to generate your project pages .
+
+Create a new file in the `src/pages` directory called `[...slug].astro` and add the following code:
+
+```astro title="src/pages/[...slug].astro"
+---
+import { useStoryblokApi } from '@storyblok/astro'
+import StoryblokComponent from '@storyblok/astro/StoryblokComponent.astro'
+import BaseLayout from '../layouts/BaseLayout.astro'
+
+export async function getStaticPaths() {
+  const storyblokApi = useStoryblokApi()
+  const { data } = await storyblokApi.get("cdn/stories", {
+    version: import.meta.env.DEV ? "draft" : "published",
+  });
+  const pages = data.stories.map(story => {
+    return {
+      params: {
+        slug: story.full_slug === 'home' ? undefined : story.full_slug
+      },
+      props: {
+        content: story.content
+      }
+    }
+  })
+  return pages
+}
+const { content } = Astro.props
+---
+<html lang="en">
+  <head>
+    <title>Storyblok and Astro</title>
+  </head>
+  <body>
+    <StoryblokComponent blok={content} />
+  </body>
+</html>
+```
+
+This file will generate a page for each story in Storyblok. It uses the `useStoryblokApi` hook to fetch all the stories and the `version` query parameter to fetch the draft version of the stories when in development mode and the published version when in building for production.
```

Comment:
```suggestion
This file will generate a page for each story, with the slug and content fetched from the Storyblok API. If the story's slug is `home`, we return an undefined slug, which generates the `/` route.
```
We don't have to repeat how the hook and query parameter works after we mentioned it earlier